frameworks/Pascal/mormot/src/raw.pas

```diff
@@ -233,6 +234,7 @@ function TRawAsyncServer.GetRawRandomWorlds(cnt: PtrInt; out res: TWorlds): bool
     pStmt.SendPipelinePrepared;
     pConn.PipelineSync;
   end;
+  pConn.Flush; // we use modified libpq what not flush inside PQPipelineSync - flush manually
```

Comment:
Sounds like a discussion that concluded by the requirement of SYNC calls. I would suggest you double check before using a modified libpq that prevents them.

Reply:
I checked. Requirements is to send a SYNC command, but not to flush client socket (do write syscall) after SYNC.
So we modify a lipbq to avoid socket flushing - see [what we did](https://github.com/pavelmash/postgres/commit/41d2dfa21ae39fe591dc0f00e97d2053b492c3cf). 
At last Just-JS do the same (send all pipeline commands in one-two socket write) and it's implementation is valid

Reply:
IMO will be forbidden to use a modified libpq.
We are testing frameworks !!

Reply:
There is a requirement what "each query must be separated by a Sync message", the problem is what in case of `libpq`  PQpipelineSync function, what add  a `S` message into buffer also flush a socket.  This is not what we need for best performance.
Form other hand `libpq` do not export public function for direct message manipulation. So we decide to do a small modification - just  [comment one line](https://github.com/pavelmash/postgres/commit/41d2dfa21ae39fe591dc0f00e97d2053b492c3cf) . 
This do not break any rules.
Many frameworks what implements PG protocol by itself also works in this way.

Reply:
I have identified the same performance issue in `h2o`, and I have started working on a proper upstream `libpq` patch, which I hope to post soon. @pavelmash, the commit in your branch is IMHO not good enough to be submitted for review because it changes the semantics of an existing public interface, which has been there since PostgreSQL 14.

The changes in this PR that concern database query pipelining do not break any rule as far as I can tell because the protocol messages are exactly the same before and after the patch. Instead, the changes reduce the number of system calls that are performed for database access, which is particularly important, since the machines used for continuous benchmarking are severely affected by speculative execution vulnerability mitigations, for example.

Reply:
@joanhey Some framework have their own PostgreSQL access layer, e.g. for asynchronous efficient process.
It is not forbidden to use another client, and other frameworks are welcome to use the patched libpq.

Reply:
Why don't send the changes to libpq?
Or why are not accepted ?

Reply:
As I said, I am working on doing that for my own version of the changes.

Reply:
And while @volyrique  working on contribution to `libpq`  we can accept this PR and verify a performance difference. This will be a good reason for Postgres team to accept such changes to trunc. In any case changing libpq in trunc is a long story...

Reply:
Not many framework uses libpq at all. But [url=https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/C%2B%2B/lithium/compile_libpq.sh]some of them[/url] patch it.

Reply:
> IMO will be forbidden to use a modified libpq. We are testing frameworks !!

Database benchmarks should be separated, we can rank the fastest databases connections and their respective languages outside HTTP Servers, this would reduce the repetitive benchmarks that basically only benchmark the better connector implementation.

With fewer database tests we should be able to add to HTTP servers benchmarks different payload sizes.

Reply:
FYI my `libpq` patch is available [here](https://www.postgresql.org/message-id/CACV6eE5arHFZEA717%3DiKEa_OewpVFfWJOmsOdGrqqsr8CJVfWQ%40mail.gmail.com).

Reply:
> FYI

anyone should use any connectors, after all, optimizations are optimizations. The bad thing is to try to compare WebFrameworks with databases instead of direct testing the database connectors/solutions.

Reply:
I don't think DB tests should be separated. Optimizations are indeed optimizations.
The real bad thing is that we compare frameworks which are proofs of concept, not usable in real-purpose code, with production ready code.
For instance, some frameworks use a direct DB connection, but only handle integer fields, so that they could shine in TFB benchmarks. Or others don't allow to change the fortunes template without re-compiling the executable - which IMHO voids the aim of a HTML template, which is to separate view and model.
I would add a requirement with some complex process instead, something with data more complex than the /fortunes test, just to validate the framework is usable for any real usecase.

Reply:
It's better to create a general issue or discussion, than talk in a PR.
We need to try to have a fair play, for all frameworks.
And later create individual issues, for each specific item.

@synopse not only the HTML template need compilation. I see some frameworks than cache the response in Fortunes, so bypassing the template, composition and html scape stage in each request.

PD: @pavelmash you can use only 1 letter server header, till is NOT added as a rule and verification.

Reply:
@joanhey - reverted by [1a48475f76c033e40fc4ddb3fa7bcbfed0bd0314] to use 1 letter, thanks!
I agree what separate issue should be created for discussion.

Hope this PR will be merged now. At last this can help Postgres team to estimate changes we propose to `libpq` what implemented by @volyrique as [libpq patch](https://www.postgresql.org/message-id/CACV6eE5arHFZEA717%3DiKEa_OewpVFfWJOmsOdGrqqsr8CJVfWQ%40mail.gmail.com). In case Postgres team decline such patch I will rollback usage of modified `libpq`